src/main/java/org/opentripplanner/routing/algorithm/strategies/connection_matrix_heuristic/Boundaries.java

```diff
@@ -23,24 +23,20 @@
         this.height = height;
         cellHeight = (latMax - latMin) / height;
         cellWidth = (lonMax - lonMin) / width;
-        latBegin = latMin - cellHeight / 2;
-        lonBegin = lonMin - cellWidth / 2;
-        latEnd = latMax + cellHeight / 2;
-        lonEnd = lonMax + cellWidth / 2;
     }
 
-    static Boundaries from(SerializedStreetHeuristicData data) {
+    static Boundaries from(SerializedConnectionMatrixHeuristicData data) {
         return new Boundaries(data.getLatMin(), data.getLatMax(), data.getLonMin(), data.getLonMax(), data.getWidth(),
                 data.getHeight());
     }
 
     Point createPointFrom(double lat, double lon) {
-        int x = (int) ((lat - latBegin) / cellHeight);
-        int y = (int) ((lon - lonBegin) / cellWidth);
+        int x = (int) ((lat - latMin) / cellHeight);
+        int y = (int) ((lon - lonMin) / cellWidth);
         return new Point(x, y);
     }
 
     boolean contains(Point point) {
-        return point.getX() >= latBegin && point.getX() <= latEnd && point.getY() >= lonBegin && point.getY() <= lonEnd;
+        return point.getI() >= 0 && point.getI() < height && point.getJ() >= 0 && point.getJ() < width;
```

Comment:
Usually `i` indicates horizontal axis, i. e. width.
E.g. i,j = 1,2 means "1 to the right, 2 up"

Reply:
As discussed, we will use descriptive `lonIndex`